etcdserver/server.go

```diff
@@ -596,7 +598,8 @@ type etcdProgress struct {
 // and helps decouple state machine logic from Raft algorithms.
 // TODO: add a state machine interface to apply the commit entries and do snapshot/recover
 type raftReadyHandler struct {
-	leadershipUpdate func()
+	leadershipUpdate   func()
+	committedIdxUpdate func(uint64)
```

Comment:
that's better. I also like to keep a consistent naming convention. should I also make `leadershipUpdate -> updateLeadership`?